runtime/lua/vim/lsp/buf.lua

```diff
@@ -143,6 +143,79 @@ local function select_client(method, on_choice)
   end
 end
 
+--- Utility function to request formatting from attached language servers. Defaults to
+--- requesting formatting from all servers attached to the current buffer.
+---
+--- @param opts table Optional table which holds the following optional fields:
+---     - formatting_options (table):
+---         Can be used to specify FormattingOptions. Some unspecified options will be
+---         automatically derived from the current Neovim options.
+---         @see https://microsoft.github.io/language-server-protocol/specification#textDocument_formatting
+---     - timeout_ms (integer, default 1000):
+---         Time in milliseconds to block for formatting requests. Formatting requests are current
+---         synchronous to prevent editing of the buffer.
+---     - bufnr (number):
+---         Restrict formatting to the clients attached to the given buffer, defaults to the current
+---         buffer (0).
+---     - filter (function):
+---         Restrict formatting to client ids returned by this function. Takes as argument the
+---         clients attached to the buffer (default to the current buffer) and returns a list
+---         of client objects from which to request formatting.
+---
+---         Example:
+---         <pre>
+---         -- Never request typescript-language-server for formatting
+---         vim.lsp.buf.format {
+---           filter = function(clients)
+---             return vim.tbl_filter(
+---               function(client) return client.name ~= "tsserver" end,
+---               clients
+---             )
+---           end
```

Comment:
```suggestion
---           filter = function(client) return client.name ~= "tsserver" end,
```

Reply:
Or do you see a special use-case where receiving the full list and returning the list of clients would have an advantage?

Reply:
The only thing I was thinking of is if someone wants to have some weird conditional fallback logic internal to the filter, like if null-ls use x if not null-ls use y

Reply:
Having access to the full list of clients could be useful in cases where a user wants to choose a server based on whether another server is attached (for example, I might want to use `eslint` in projects that have it configured but fall back to `tsserver` otherwise). I don't mind this either way but wanted to mention a potential use case there.

Reply:
That was my motivation for keeping them separate

Reply:
Okay makes sense. But in that case you'll have to revert part of the doc changes I had proposed because it already changed the filter description